src/server/dcp_server/models.py

```diff
@@ -60,12 +75,475 @@ def masks_to_outlines(self, mask):
 
+        return loss
+    
+    def eval(self, imgs):
+    ## TODO should call forward once, model is in eval mode, and return predicted masks
+        """
+        Evaluate the model on the provided images and return predicted labels.
+            Input:
+            imgs: List[np.ndarray[np.uint8]] with shape (3, dx, dy)
+            Output:
+            labels: List of predicted labels.
+        """
+        labels = []
+        for img in imgs:
+            
+            img = torch.from_numpy(img).unsqueeze(0)
+            labels.append(self.forward(img))
+
+        return labels
 # class CustomSAMModel():
 # # https://github.com/facebookresearch/segment-anything/blob/main/notebooks/automatic_mask_generator_example.ipynb
 #     def __init__(self):
 #         pass
 
 
+class CellposePatchCNN():
+
+    """
+    Cellpose & patches of cells and then cnn to classify each patch
+    """
+    
+    def __init__(self,  
+    segmentor_model_config, segmentor_train_config, segmentor_eval_config,
+    classifier_model_config, classifier_train_config, classifier_eval_config ):
+
+        
+        self.segmentor_model_config = segmentor_model_config
+        self.segmentor_train_config = segmentor_train_config
+        self.segmentor_eval_config = segmentor_eval_config
+
+        self.classifier_model_config = classifier_model_config
+        self.classifier_train_config = classifier_train_config
+        self.classifier_eval_config = classifier_eval_config
+
+        # Initialize the classifier and the cellpose model 
+        self.classifier = CellFullyConvClassifier(
+            classifier_model_config, classifier_train_config,  classifier_eval_config)
+        self.segmentor = CustomCellposeModel(
+            segmentor_model_config, segmentor_train_config, segmentor_eval_config)
+
+    def init_from_checkpoints(self, chpt_classifier=None, chpt_segmentor=None):
+        """
+        Initialize the model from pre-trained checkpoints.
+        """
+
+        self.segmentor = CustomCellposeModel(
+            model_config={"gpu":torch.cuda.is_available(), "pretrained_model":chpt_segmentor} 
+            )
+        self.classifier.load_state_dict(torch.load(chpt_classifier)["model"])
+
+
+    def train(self, imgs, masks):
+
+        # masks should have first channel as a cellpose mask and
+        # all other layers corresponds to the classes
+        ## TODO: take care of the images and masks preparation -> this step isn't for now @Mariia
+        ## TODO call create_patches (adjust create_patch_dataset function)
+        ## to prepare imgs and masks for training CNN
+        ## TODO call self.classifier.train(imgs, masks)
+
+        num_classes = self.classifier_model_config.get("num_classes", 3)
+
+        black_bg = self.classifier_model_config.get("black_bg", False)
+        include_mask = self.classifier_model_config.get("include_mask", False)
+        include_cellpose_mask = self.classifier_model_config.get("include_cellpose_mask", True)
```

Comment:
I can't find where in the code these are used: black_bg, include_mask and include_cellpose_mask -> remove?

Reply:
We used it in the beginning, but I save it just in case we want to experiment. I remove it.

Reply:
yes please remove whatever is not used, thanks!